bundles/org.openhab.binding.caddx/src/main/java/org/openhab/binding/caddx/internal/handler/CaddxBridgeHandler.java

```diff
@@ -0,0 +1,408 @@
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.openhab.binding.caddx.internal.CaddxBindingConstants;
+import org.openhab.binding.caddx.internal.CaddxCommunicator;
+import org.openhab.binding.caddx.internal.CaddxEvent;
+import org.openhab.binding.caddx.internal.CaddxMessage;
+import org.openhab.binding.caddx.internal.CaddxMessageType;
+import org.openhab.binding.caddx.internal.CaddxPanelListener;
+import org.openhab.binding.caddx.internal.CaddxProtocol;
+import org.openhab.binding.caddx.internal.CaddxSource;
+import org.openhab.binding.caddx.internal.config.CaddxBridgeConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxKeypadConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxPartitionConfiguration;
+import org.openhab.binding.caddx.internal.config.CaddxZoneConfiguration;
+import org.openhab.binding.caddx.internal.discovery.CaddxDiscoveryService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The bridge handler for the Caddx RS232 Serial interface.
+ *
+ * @author Georgios Moutsos - Initial contribution
+ */
+@NonNullByDefault
+public class CaddxBridgeHandler extends BaseBridgeHandler implements CaddxPanelListener {
+    private final Logger logger = LoggerFactory.getLogger(CaddxBridgeHandler.class);
+
+    static final byte[] DISCOVERY_PARTITION_STATUS_REQUEST_0 = { 0x26, 0x00 };
+    static final byte[] DISCOVERY_ZONES_SNAPSHOT_REQUEST_00 = { 0x25, 0x00 };
+    static final byte[] DISCOVERY_ZONES_SNAPSHOT_REQUEST_10 = { 0x25, 0x10 };
+    static final byte[] DISCOVERY_ZONES_SNAPSHOT_REQUEST_20 = { 0x25, 0x20 };
+    static final byte[] DISCOVERY_PARTITIONS_SNAPSHOT_REQUEST = { 0x27 };
+
+    private final SerialPortManager portManager;
+    private @Nullable CaddxDiscoveryService discoveryService = null;
+    private CaddxProtocol protocol = CaddxProtocol.Binary;
+    private String serialPortName = "";
+    private int baudRate;
+    private @Nullable CaddxCommunicator communicator = null;
+
+    public @Nullable CaddxDiscoveryService getDiscoveryService() {
+        return discoveryService;
+    }
+
+    public void setDiscoveryService(CaddxDiscoveryService discoveryService) {
+        this.discoveryService = discoveryService;
+    }
+
+    /**
+     * Constructor.
+     *
+     * @param bridge
+     */
+    public CaddxBridgeHandler(SerialPortManager portManager, Bridge bridge) {
+        super(bridge);
+
+        this.portManager = portManager;
+    }
+
+    private void init() {
```

Comment:
Any reason this code isn't in your `initialize` method?

Reply:
No. 
Just a leftover from various rewrites of the function.
Moved to the initialize function.